File: frontend/src/types/reports.ts
export interface SpendingByCategory {
  category_id: string | null;
  category_name: string;
  amount: string;
  percentage: string;
}

export interface SpendingReport {
  from_date: string;
  to_date: string;
  currency: string;
  total: string;
  categories: SpendingByCategory[];
}

export interface IncomeVsExpensePeriod {
  period: string;
  income: string;
  expenses: string;
  net: string;
}

export interface IncomeVsExpenseReport {
  from_date: string;
  to_date: string;
  currency: string;
  periods: IncomeVsExpensePeriod[];
  total_income: string;
  total_expenses: string;
  total_net: string;
}

export interface MonthlyTrend {
  month: string;
  income: string;
  expenses: string;
  net: string;
}

export interface TrendsReport {
  months: number;
  currency: string;
  trends: MonthlyTrend[];
}

export interface AccountBalance {
  account_id: string;
  account_name: string;
  account_type: string;
  currency: string;
  balance: string;
  balance_base: string;
}

export interface NetWorthReport {
  currency: string;
  date: string;
  accounts: AccountBalance[];
  total_assets: string;
  total_liabilities: string;
  net_worth: string;
}

File: backend/app/schemas/reports.py
import uuid
from datetime import date
from decimal import Decimal

from pydantic import BaseModel


class SpendingByCategory(BaseModel):
    category_id: uuid.UUID | None
    category_name: str
    amount: Decimal
    percentage: Decimal


class SpendingReportResponse(BaseModel):
    from_date: date
    to_date: date
    currency: str
    total: Decimal
    categories: list[SpendingByCategory]


class IncomeVsExpensePeriod(BaseModel):
    period: str  # e.g. "2024-01" for monthly
    income: Decimal
    expenses: Decimal
    net: Decimal


class IncomeVsExpenseResponse(BaseModel):
    from_date: date
    to_date: date
    currency: str
    periods: list[IncomeVsExpensePeriod]
    total_income: Decimal
    total_expenses: Decimal
    total_net: Decimal


class MonthlyTrend(BaseModel):
    month: str  # "YYYY-MM"
    income: Decimal
    expenses: Decimal
    net: Decimal


class TrendsReportResponse(BaseModel):
    months: int
    currency: str
    trends: list[MonthlyTrend]


class AccountBalance(BaseModel):
    account_id: uuid.UUID
    account_name: str
    account_type: str
    currency: str
    balance: Decimal
    balance_base: Decimal


class NetWorthResponse(BaseModel):
    currency: str
    date: date
    accounts: list[AccountBalance]
    total_assets: Decimal
    total_liabilities: Decimal
    net_worth: Decimal
